api/v3/VolunteerUtil.php

```diff
@@ -143,13 +143,17 @@ function civicrm_api3_volunteer_util_getsupportingdata($params) {
 
     $results['phone_types'] = CRM_Core_OptionGroup::values("phone_type", FALSE, FALSE, TRUE);
 
-    $results['defaults'] = array(
-      'profile' => civicrm_api3('UFGroup', 'getvalue', array(
-        "name" => "volunteer_sign_up",
-        "return" => "id"
-      )),
-      'relationships' => _volunteerGetProjectRelationshipDefaults(),
-    );
+    //Fetch the Defaults from saved settings.
+    $defaults = volunteer_composeDefaultArray();
+
+    //StopGap because the interface for contacts didn't fit into scope
+    if(!array_key_exists("relationships", $defaults)) {
+     $defaults['relationships'] = _volunteerGetProjectRelationshipDefaults();
+    }
+    //Allow other extensions to modify the defaults
+    CRM_Volunteer_Hook::runDefaultSettingsHook($defaults);
+
+    $results['defaults'] = $defaults;
```

Comment:
So, I'm having misgivings about the use of the _defaults_ namespace -- not just here but in general (e.g., setting names). It's a little broad.

In the case of this API, is there a chance we'd return different defaults, or defaults for a different object altogether, depending on the value of `$params['controller']`? How do the answers to those questions inform our opinion of the appropriateness of calling the result `$results['defaults']`?

In the case of the settings themselves, mightn't we in the future decide we also need configurable defaults for needs? In that case, wouldn't a naming scheme like `volunteer_default_is_active` introduce the likelihood of naming collisions... or at least confusion?